### ISSUE TITLE
Triggering backup from outside the pod

### ISSUE THREAD

**Issue by KarlisAG** [zalando/postgres-operator#2487]
Please, answer some short questions which should help us to understand your problem / question better?

- **Which image of the operator are you using?** e.g. registry.opensource.zalan.do/acid/postgres-operator:v1.10.1
- **Where do you run it - cloud or metal? Kubernetes or OpenShift?** AKS
- **Are you running Postgres Operator in production?** yes
- **Type of issue?** Question

I am wondering is it possible to manage postgres backups outside the pods? I know there are scheduled backups that we use and work fine. But there cases where we would want to create a backup on demand, just for some safety reasons. Based on documentation we have exec'ed into the container and ran the backup command. It has done its job well. But we would love to avoid having to exec into the pods, especially in production. Ideally we could trigger it outside of the pod. If we made separate pod that did it then it would still kind of have to exec so that is not an option. I checked the UI and it also doesn't seem to support backups. Only doable option seemed to add a new sidecar that would act as an API that would just react to requests, which we would send from outside, and if request came in with correct token then it would make a backup. First of all, that is pretty hacky, secondly it might not even work as expected - not sure if sidecar can even run the same backup scripts and have the same and working result.

So I have a question to the community - have you faced a similar issue, is there already something available for this that we just missed, maybe you have some workaround or alternatives to this?

**Comment by FxKu** [comment]
Have you tried using a K8s job resource to perform backups. I also like the idea of using a extra CRD to manage backups as suggested in #2141

**Comment by KarlisAG** [comment]
Agreed, the CRD I think also would be the best solution. At least then we can be sure that it is done properly and operator can manage it.

As for the job - no, haven't tried that yet. It was one of the ideas we had but I am having hard time visualising how that would work. My assumption was that for the backup the script had to be ran and then the container needs to have the direct access to the postgres cluster and configuration it has from the operator (this is why I thought that sidecar would be the better approach)? Can backup be made outside of the pod without exec and without sharing the same storage? I would assume that regular pg_dump wouldn't work as we use Azure and we want the backup to end up in the Azure Storage Account as well so that we can clone, if it is required. Or are my assumptions wrong on all these accounts? Please let me know.
